Spanner/tests/System/TransactionTest.php

```diff
@@ -217,6 +221,76 @@ public function testStrongRead()
         $this->assertInstanceOf(Timestamp::class, $snapshot->readTimestamp());
     }
 
+    public function testRunTransactionWithDatabaseRole()
+    {
+        // Emulator does not support FGAC
+        $this->skipEmulatorTests();
+        
+        $db = self::$databaseWithRestrictiveDatabaseRole;
+
+        $row = $this->getRow();
+        $row['name'] = 'Doug';
+
+        $db->runTransaction(function ($t) use ($row) {
+            $t->update('Users', $row);
+
+            $t->commit();
+        });
+        $row = $this->getRow();
+        $this->assertEquals('Doug', $row['name']);
+
+        try {
+            $db->runTransaction(function ($t) {
+                $id = rand(1, 346464);
+                $t->insert(self::TEST_TABLE_NAME, [
+                    'id' => $id,
+                    'name' => uniqid(self::TESTING_PREFIX),
+                    'birthday' => new Date(new \DateTime)
+                ]);
+
+                $t->commit();
+            });
+        } catch (ServiceException $e) {
+            $this->assertInstanceOf(ServiceException::class, $e);
```

Comment:
Test should fail if no exception is thrown

Reply:
Done